lib/membrane_http_adaptive_stream/source/client_genserver.ex

```diff
@@ -6,28 +6,36 @@ defmodule Membrane.HTTPAdaptiveStream.Source.ClientGenServer do
   use GenServer
   alias ExHLS.Client
 
-  @spec start_link(
-          String.t(),
-          Membrane.HTTPAdaptiveStream.Source.variant_selection_policy(),
-          Membrane.Time.t()
-        ) ::
+  @spec start_link(%{
+          url: String.t(),
+          variant_selection_policy: Membrane.HTTPAdaptiveStream.Source.variant_selection_policy(),
+          source: pid(),
+          how_much_to_skip: Membrane.Time.t()
+        }) ::
           GenServer.on_start()
-  def start_link(url, variant_selection_policy, how_much_to_skip) do
+  def start_link(%{
+        url: url,
+        variant_selection_policy: variant_selection_policy,
+        source: source,
+        how_much_to_skip: how_much_to_skip
+      }) do
     GenServer.start_link(__MODULE__,
       url: url,
       variant_selection_policy: variant_selection_policy,
+      source: source,
       how_much_to_skip: how_much_to_skip
     )
   end
 
-  @spec request_audio_chunk(pid()) :: :ok
-  def request_audio_chunk(client_genserver) do
-    GenServer.cast(client_genserver, {:request_audio_chunk, self()})
+  @spec request_chunk_or_eos(pid()) :: :ok
+  def request_chunk_or_eos(client_genserver) do
+    GenServer.cast(client_genserver, {:request_chunk_or_eos, self()})
   end
 
-  @spec request_video_chunk(pid()) :: :ok
-  def request_video_chunk(client_genserver) do
-    GenServer.cast(client_genserver, {:request_video_chunk, self()})
+  @spec get_skipped_segments_cumulative_duration_ms(pid()) ::
+          {:ok, Membrane.Time.t()} | {:error, reason :: any()}
```

Comment:
I don't think `GenServer.call()` returns `{:ok, <result>}` tuple.
Another thing is that since we return result represented as `Membrane.Time.t()`, I don't think we should use `_ms` in the function's name